src/java.base/share/classes/java/lang/Class.java

```diff
@@ -706,7 +706,7 @@ private static Class<?> forName(Module module, String name, Class<?> caller) {
      */
     @PreviewFeature(feature = PreviewFeature.Feature.VALUE_OBJECTS)
     public boolean isValue() {
-        return (this.getModifiers() & Modifier.VALUE) != 0;
+        return (this.getModifiers() & Modifier.IDENTITY) == 0;
```

Comment:
This needs to test if it's a primitive type, `void`, and an array type which do not have `IDENTITY` modifier set.